ibis/backends/datafusion/__init__.py

```diff
@@ -526,7 +514,8 @@ def to_pyarrow_batches(
         schema = table_expr.schema()
         names = schema.names
 
-        struct_schema = schema.as_struct().to_pyarrow()
+        nullable_schema = schema.as_nullable()
```

Comment:
DataFusion has what I would consider a bug in how it determines the types of column in select statements. Sometimes columns are inferred as not nullable, but that should effectively never be true for a `SELECT` statement, as pretty much **any** operation can introduce a `NULL`.